tests/unit/test_file_system.py

```diff
@@ -1,97 +1,205 @@
-import sys
-from pathlib import Path
-from stat import S_IRWXO
-from typing import TYPE_CHECKING
-
+import os
+import pathlib
+import threading
+from collections.abc import AsyncGenerator, Generator
+from http.server import HTTPServer
+from typing import Any
+
+import aiohttp
 import pytest
+from fsspec.implementations.http import HTTPFileSystem
 from fsspec.implementations.local import LocalFileSystem
+from RangeHTTPServer import RangeRequestHandler  # type: ignore[import-untyped]
+
+from litestar.file_system import (
+    BaseFileSystem,
+    BaseLocalFileSystem,
+    FileSystemRegistry,
+    FsspecAsyncWrapper,
+    FsspecSyncWrapper,
+    maybe_wrap_fsspec_file_system,
+)
+from litestar.testing.client.subprocess_client import _get_available_port
+
+
+class PatchedHTTPFileSystem(HTTPFileSystem):  # type: ignore[misc]
+    async def _cat_file(self, *args: Any, **kwargs: Any) -> Any:
+        try:
+            return await super()._cat_file(*args, **kwargs)
+        except aiohttp.ClientResponseError as exc:
+            # this is an implementation specific error case; we expect most file systems
+            # to return an empty byte-string when trying to read beyond the limits of
+            # the file
+            if exc.status == 416:
+                return b""
+            raise exc
+
+
+@pytest.fixture(scope="session")
+def tmp_dir(tmp_path_factory: pytest.TempPathFactory) -> pathlib.Path:
+    return tmp_path_factory.mktemp("test")
+
+
+@pytest.fixture(scope="session")
+def http_server_port() -> int:
+    return _get_available_port()
+
+
+@pytest.fixture(scope="session")
+def file_path(tmp_dir: pathlib.Path) -> pathlib.Path:
+    path = tmp_dir / "test.txt"
+    path.write_bytes(b"0123456789")
+    return path
+
+
+@pytest.fixture(scope="session")
+def file(file_path: pathlib.Path) -> pathlib.Path:
+    return pathlib.Path(file_path.name)
+
+
+@pytest.fixture()
+def local_fs(monkeypatch: pytest.MonkeyPatch, file_path: pathlib.Path) -> BaseLocalFileSystem:
+    monkeypatch.chdir(file_path.parent)
+    return BaseLocalFileSystem()
+
+
+@pytest.fixture()
+def fsspec_local_fs(monkeypatch: pytest.MonkeyPatch, file_path: pathlib.Path) -> BaseFileSystem:
+    monkeypatch.chdir(file_path.parent)
+    return FsspecSyncWrapper(LocalFileSystem())
```

Comment:
dont we miss the async wrapper in the fixtures ?

Reply:
The `http_fs` fixture has it